spec/support/backups_helpers.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module BackupsHelpers
+  def setup_local_backups
+    root_directory = Dir.mktmpdir
+    SiteSetting.backup_location = BackupLocationSiteSetting::LOCAL
+    root_directory
+  end
+
+  def teardown_local_backups(root_directory:)
```

Comment:
I wonder if we should just provide a `with_local_backup_files` method which yields the block so that we can handle deleting the directory automatically. Something like Ruby's `Tempfile.open`

```
Tempfile.open('foo', '/home/temp') do |f|
   ... do something with f ...
end
```

Reply:
I get what you mean but how would you use this nicely within the `it` block? The reason we need to create these backups is so this table is populated

![image](https://github.com/user-attachments/assets/56b6853a-53e1-4df2-90a7-f03ba5ca6585)

It would be tricky to have to do all the file creation and test assertions etc within the `with_local_backup_files` block

Reply:
> how would you use this nicely within the `it` block?

Good point, I'm don't think there is a nice way to do so.